### DISCUSSION TITLE
X as sparse tensor in Data

### DISCUSSION THREAD

**Discussion by niknoproblems** [pyg-team/pytorch_geometric#2501] [Q&A]
Is it possible to use sparse tensor for X?

**Comment by rusty1s** [comment]
It depends. In general yes. However, you might need to modify mini-batch mode behavior since `SparseTensor`s are generally stacked diagonally instead of row-wise in `DataLoader`.

**Reply by A11en0** [reply]
How to modify it? could you please give an example?

**Reply by rusty1s** [reply]
```python
class MyData(Data):
    def __cat_dim__(self, key, value):
        if key == 'x':
            return 0
        return super().__cat_dim__(key, value)
```

**Reply by A11en0** [reply]
thanks!

**Reply by rusty1s** [reply]
Is your last comment resolved? I cannot find it anymore :)

**Reply by A11en0** [reply]
yes, I solved it! Thanks!
